mne/preprocessing/xdawn.py

```diff
@@ -222,34 +222,25 @@ def __init__(self, n_components=2, signal_cov=None, correct_overlap='auto',
         self.patterns_ = dict()
         self.evokeds_ = dict()
 
-        if correct_overlap not in ['auto', True, False]:
-            raise ValueError('correct_overlap must be a bool or "auto"')
-        self.correct_overlap = correct_overlap
-
-    def fit(self, epochs, y=None):
+    def fit(self, X, y=None):
         """Fit Xdawn from epochs.
 
         Parameters
         ----------
-        epochs : Epochs object
-            An instance of Epoch on which Xdawn filters will be trained.
-        y : ndarray | None (default None)
-            Not used, here for compatibility with decoding API.
+        X : ndarray, shape(n_channels, n_times * n_freq)
+        y : ndarray shape(n_samples,) | None (default None)
+            labels for the data
 
         Returns
         -------
         self : Xdawn instance
             The Xdawn instance.
         """
-        if self.correct_overlap == 'auto':
-            self.correct_overlap = _check_overlapp(epochs)
-
+        epochs_data = X.reshape(X.shape[0], info['nchan'], X.shape[1] / 
+                                            info['nchan'])
         # Extract signal covariance
         if self.signal_cov is None:
-            if self.correct_overlap:
-                sig_data = _construct_signal_from_epochs(epochs)
-            else:
-                sig_data = np.hstack(epochs.get_data())
+            sig_data = np.hstack(epochs_data)
```

Comment:
Can you add something in mne/doc/whatsnew this is a significant change of xdawn behavior